pyanalyze/name_check_visitor.py

```diff
@@ -276,6 +278,19 @@ def _not_in(a: object, b: Container[object]) -> bool:
 }
 
 
+class CustomContextManager(Protocol[T, U]):
+    def __enter__(self) -> T:
+        raise NotImplementedError
+
+    def __exit__(
+        self,
+        _exc_type: Optional[Type[BaseException]],
+        _exc_value: Optional[BaseException],
```

Comment:
```suggestion
        __exc_type: Optional[Type[BaseException]],
        __exc_value: Optional[BaseException],
```
This marks the parameters as positional-only; otherwise pyanalyze enforces that the names match.